src/index.test.ts

```diff
@@ -223,6 +253,160 @@ describe('Geckoboard', () => {
     );
   });
 
+  it.each([
+    {
+      datetime: new Date('2018'),
+      expected: '2018-01-01T00:00:00.000Z',
+      description: 'just a year string',
+    },
+    {
+      datetime: new Date('2018-02'),
+      expected: '2018-02-01T00:00:00.000Z',
+      description: 'just a month string',
+    },
+    {
+      datetime: new Date('2018-03-02'),
+      expected: '2018-03-02T00:00:00.000Z',
+      description: 'just a date string',
+    },
+    {
+      datetime: new Date('2018-04-05T12:00:00.000Z'),
+      expected: '2018-04-05T12:00:00.000Z',
+      description: 'a UTC datetime string',
+    },
+    {
+      datetime: new Date('2013-09-15T05:53:00+08:00'),
+      expected: '2013-09-14T21:53:00.000Z',
+      description: 'a datetime string with timezone information (+ positve)',
+    },
+    {
+      datetime: new Date('2018-01-12T20:00:00-10:00'),
+      expected: '2018-01-13T06:00:00.000Z',
+      description: 'a datetime string with timezone information (- negative)',
+    },
+    {
+      datetime: new Date('2018-01-12T00:00:01'),
+      expected: '2018-01-11T23:00:01.000Z',
+      description: 'a datetime string in local time (mocked to `Europe/Paris`)',
+    },
+  ])(
+    'can use date object that is constructed with $description that will be converted to UTC to append datetime data',
+    async ({ datetime, expected }) => {
+      const serverSpy = jest.fn().mockReturnValue('{}');
+      mockPool
+        .intercept({
+          method: 'POST',
+          path: '/datasets/steps.by.day/data',
+        })
+        .reply(200, serverSpy);
+      const dataset = prepareDataset();
+
+      await dataset.append([
+        {
+          timestamp: datetime,
+          steps: 8190,
+        },
+      ]);
+      expect(serverSpy).toHaveBeenCalledWith(
+        expect.objectContaining({
+          method: 'POST',
+          path: '/datasets/steps.by.day/data',
+          headers: expect.objectContaining({
+            authorization: `Basic ${btoa('API_KEY:')}`,
+            'user-agent': 'Geckoboard Node Client 2.0.0',
+            'content-type': 'application/json',
+          }),
+          body: JSON.stringify({
+            data: [
+              {
+                timestamp: expected,
+                steps: 8190,
+              },
+            ],
+          }),
+        }),
+      );
+    },
+  );
+
+  it.each([
+    {
```

Comment:
🏄 worth extracting these cases so they're automatically shared across the scenarios?